network/wsPeer.go

```diff
@@ -944,6 +950,22 @@ func (wp *wsPeer) sendMessagesOfInterest(messagesOfInterestGeneration uint32, me
 	}
 }
 
+func (wp *wsPeer) IdentityVerified() {
+	wp.identityVerified = true
```

Comment:
probalby make identityVerified a uint32 and use sync/atomic operations to set and test it here and in waitForIdentityVerify()